File: problems/HackerRank/2521_exercises/linkedList/intermediate/deleteDuplicates.c
struct ListNode *deleteDuplicates(struct ListNode *head)
{

	if (head == NULL)
	{
		return head;
	}

	struct ListNode *dummyNode = (struct ListNode *)malloc(sizeof(struct ListNode));
	dummyNode->next = head;
	struct ListNode *slow = dummyNode;
	struct ListNode *current = head;

	while (current != NULL)
	{
		// check duplicate nodes
		if (current->next && current->val == current->next->val)
		{
			// move the point unitl find new node
			while (current->next && current->val == current->next->val)
			{
				current = current->next;
			}
			slow->next = current->next;
		}
		else
		{
			slow = slow->next;
		}
		current = current->next;
	}
	return dummyNode->next;
}

File: problems/LeetCode/linked_list/deleteDuplicates.py
def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
		if not head: return None

		cur = head

		while cur.next:
				if cur.next.val == cur.val:
						cur.next = cur.next.next
				else:
						cur = cur.next
		
		return head